src/Bundle/Grid/AbstractGrid.php

```diff
@@ -15,9 +15,34 @@
 
 use Sylius\Bundle\GridBundle\Builder\GridBuilder;
 use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Component\Grid\Attribute\AsGrid;
 
 abstract class AbstractGrid implements GridInterface
 {
+    public static function getName(): string
+    {
+        if ($attribute = self::getAttributes()) {
+            if (!empty($attribute[0])) {
+                return $attribute[0]->newInstance()->name ?? static::class;
+            }
+        }
+
+        return static::class;
+    }
+
+    public function getResourceClass(): string
+    {
+        if ($attribute = self::getAttributes()) {
+            return $attribute[0]->newInstance()->resourceClass;
+        }
+
+        throw new \LogicException(sprintf(
+            'You have to implement %s method or use %s attribute.',
+            __METHOD__,
+            AsGrid::class,
+        ));
+    }
```

Comment:
Can we use string parameter instead of the class? It can be useful for plugins.

Reply:
I'm not sure what you are meaning, which line ?